iceoryx_examples/request_response/README.md

```diff
@@ -0,0 +1,217 @@
+        });
+```
+
+In the same while loop the client receives the responses from the server using `take()`
+and extract the sequence id with `response.getResponseHeader().getSequenceId()`.
+When the server response comes in the correct order, the received sum is re-used to
+insert it to the `addend` of the next request to send.
+
+<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [[take response]] -->
+```cpp
+    while (client.take().and_then([&](const auto& response) {
+        auto receivedSequenceId = response.getResponseHeader().getSequenceId();
+        if (receivedSequenceId == guardedCtx->expectedResponseSequenceId)
+        {
+            guardedCtx->fibonacciLast = guardedCtx->fibonacciCurrent;
+            guardedCtx->fibonacciCurrent = response->sum;
+            std::cout << APP_NAME << " Got Response : " << guardedCtx->fibonacciCurrent << std::endl;
+        }
+        else
+        {
+            std::cout << "Got Response with outdated sequence ID! Expected = "
+                      << guardedCtx->expectedResponseSequenceId << "; Actual = " << receivedSequenceId
+                      << "! -> skip" << std::endl;
+        }
+    }))
+    {
+    }
+```
+
+### Server using Listener
+
+At first, the includes for the server port and request-response types and runtime are needed.
+<!-- [geoffrey] [geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp] [iceoryx includes] -->
+```cpp
+#include "request_and_response_types.hpp"
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/popo/notification_callback.hpp"
+#include "iceoryx_posh/popo/server.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+```
+
+First, a callback is created that shall be called when server receives a response.
+In this case the calculation is here done too but more resource consuming tasks
+should be maybe outsourced.
+
+<!--[geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp][request callback]-->
+```cpp
+void onRequestReceived(iox::popo::Server<AddRequest, AddResponse>* server)
+{
+    while (server->take().and_then([&](const auto& request) {
+        std::cout << APP_NAME << " Got Request: " << request->augend << " + " << request->addend << std::endl;
+
+        server->loan(request)
+            .and_then([&](auto& response) {
+                response->sum = request->augend + request->addend;
+                std::cout << APP_NAME << " Send Response: " << response->sum << std::endl;
+                response.send();
+            })
+            .or_else([](auto& error) {
+                std::cout << "Could not allocate Response! Return value = " << static_cast<uint64_t>(error)
+                          << std::endl;
+            });
+    }))
+    {
+    }
+}
+```
+
+From the server port is the `take()` API used to receive the data and then the `loan()` and send `send()` calls
```

Comment:
Maybe: The server provides the `take()` method for receiving requests and the `loan()` and `send()` methods for sending the responses with the sum of the two numbers